wrapper/src/test/java/integration/refactored/host/TestEnvironment.java

```diff
@@ -0,0 +1,669 @@
+      } catch (ClassNotFoundException e) {
+        throw new RuntimeException(
+            "Driver not found: "
+                + DriverHelper.getDriverClassname(env.info.getRequest().getDatabaseEngine()),
+            e);
+      }
+
+      final String url =
+          String.format(
+              "%s%s:%d/%s",
+              DriverHelper.getDriverProtocol(env.info.getRequest().getDatabaseEngine()),
+              env.info.getDatabaseInfo().getClusterEndpoint(),
+              env.info.getDatabaseInfo().getClusterEndpointPort(),
+              env.info.getDatabaseInfo().getDefaultDbName());
+
+      try {
+        env.auroraUtil.addAuroraAwsIamUser(
+            env.info.getRequest().getDatabaseEngine(),
+            url,
+            env.info.getDatabaseInfo().getUsername(),
+            env.info.getDatabaseInfo().getPassword(),
+            env.info.getIamUsername(),
+            env.info.getDatabaseInfo().getDefaultDbName());
+
+      } catch (SQLException e) {
+        throw new RuntimeException("Error configuring IAM access.", e);
+      }
+    }
+  }
+
+  private static String getEnvironmentInfoAsString(TestEnvironment env) {
+    try {
+      final ObjectMapper mapper = new ObjectMapper();
+      return mapper.writeValueAsString(env.info);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Error serializing environment details.", e);
+    }
+  }
+
+  public void runTests(String taskName) throws IOException, InterruptedException {
+    final ContainerHelper containerHelper = new ContainerHelper();
+    containerHelper.runTest(this.testContainer, taskName);
+  }
+
+  public void debugTests(String taskName) throws IOException, InterruptedException {
+    final ContainerHelper containerHelper = new ContainerHelper();
+    containerHelper.debugTest(this.testContainer, taskName);
+  }
+
+  @Override
+  public void close() throws Exception {
+    if (this.databaseContainers != null) {
+      for (GenericContainer<?> container : this.databaseContainers) {
+        try {
+          container.stop();
+        } catch (Exception ex) {
+          // ignore
+        }
+      }
+      this.databaseContainers.clear();
+    }
+
+    if (this.testContainer != null) {
+      this.testContainer.stop();
+      this.testContainer = null;
+    }
+
+    if (this.proxyContainers != null) {
+      for (ToxiproxyContainer proxyContainer : this.proxyContainers) {
+        proxyContainer.stop();
+      }
+      this.proxyContainers = null;
+    }
+
+    switch (this.info.getRequest().getDatabaseEngineDeployment()) {
+      case AURORA:
+        deleteAuroraDbCluster();
+        break;
+      case RDS:
+        throw new NotImplementedException(this.info.getRequest().getTargetJvm().toString());
+      default:
+        // do nothing
+    }
+  }
+
+  private void deleteAuroraDbCluster() {
+    if (!StringUtils.isNullOrEmpty(this.runnerIP)) {
+      auroraUtil.ec2DeauthorizesIP(runnerIP);
```

Comment:
if we are running the integration tests locally, does this deauthorize the IP for our machine or the IP of the test container? If its the IP of our machine, do we want to make sure it isn't deauthorized if we're running locally so that we can still connect to aurora directly after running tests?